sklearn/base.py

```diff
@@ -602,6 +602,42 @@ def _validate_data(
 
         return out
 
+    def _check_features_support(self, X, *, delegate=None, reset=True):
+        """Set or check both `n_features_in_` and `feature_names_in_` based on delegate.
```

Comment:
Since `_check_features_support` might be a bit generic for people not aware to what we intend to do, would it be good to reference the SLEP of the `n_features_in_` and `feature_names_in_` as well (in the long description of the docstring).